### DISCUSSION TITLE
Peridigm contact model

### DISCUSSION THREAD

**Discussion by Martin1330** [peridigm/peridigm#243] [Q&A] [answered]
Hello community,
In our contact model, in some Dr. Silling papers, he chosed the contact radius dpi = min{0.9‖x′ − x‖, 1.35|Δx| }. However, in our peridigm, it was chosen a fixed value. I saw your suggestions"Initial guess for contact radius: somewhat smaller than the initial node spacing,
so that it doesn't add forces to material in the undeformed or slightly deformed configuration. I would also expect it to be large enough to prevent material interpenetration." 
I'd like to know whether these two methods have no big effect? Do you recommend the latter first？ Thank you so much.
Best regards,
Martin

**Comment by sahirbutt** [comment, marked as answer]
In my experience, for two bodies in contact (A and B) with a constant mesh size (radius) dxa and dxb, I pick the contact radius to be (dxa+dxb). This avoids the contact forces in the undeformed configuration.
However, if the mesh size is not constant for whole body (A or B), then it becomes problematic. For that case, I find a unique contact radius for every contacting pair of points in Peridigm_ShortRangeForceContactModel.cpp (I did it using the volume to compute the radius of the pair of points, as I remember having some issues to access the Radius directly in the ContactBlock).

**Reply by Martin1330** [reply]
Thank you so much for sharing these valuable experiences.
